h2o-algos/src/test/java/hex/tree/TreeHandlerTest.java

```diff
@@ -0,0 +1,219 @@
+package hex.tree;
+
+import hex.genmodel.algos.tree.SharedTreeNode;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.tree.gbm.GBM;
+import hex.tree.gbm.GBMModel;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.*;
+import water.fvec.Frame;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.*;
+
+public class TreeHandlerTest extends TestUtil {
+
+    @BeforeClass
+    public static void stall() {
+        stall_till_cloudsize(1);
+    }
+
+    @Test
+    public void testSharedTreeSubgraphConversion() {
+        Frame tfr = null;
+        GBMModel model = null;
+
+        Scope.enter();
+        try {
+            tfr = parse_test_file("./smalldata/airlines/allyears2k_headers.zip");
+            DKV.put(tfr);
+            GBMModel.GBMParameters parms = new GBMModel.GBMParameters();
+            parms._train = tfr._key;
+            parms._response_column = "Dest";
+            parms._ntrees = 1;
+            parms._seed = 0;
+
+            model = new GBM(parms).trainModel().get();
+            final SharedTreeModel.SharedTreeOutput sharedTreeOutput = model._output;
+            assertEquals(parms._ntrees, sharedTreeOutput._ntrees);
+            assertEquals(parms._ntrees, sharedTreeOutput._treeKeys.length);
+            assertEquals(parms._ntrees, sharedTreeOutput._treeKeysAux.length);
+
+            final int treeNumber = 0;
+            final int treeClass = 0;
+            final CompressedTree auxCompressedTree = sharedTreeOutput._treeKeysAux[treeNumber][treeClass].get();
+            final SharedTreeSubgraph sharedTreeSubgraph = sharedTreeOutput._treeKeys[treeNumber][treeClass].get()
+                    .toSharedTreeSubgraph(auxCompressedTree, sharedTreeOutput._names, sharedTreeOutput._domains);
+
+            assertNotNull(sharedTreeSubgraph);
+
+            final TreeHandler.CompressedTreeFormat compressedTreeFormat = TreeHandler.convertSharedTreeSubgraph(sharedTreeSubgraph);
+            assertNotNull(compressedTreeFormat);
+
+            assertEquals(0, compressedTreeFormat.getRootNodeNumber());
+
+            // Test against manually walked tree to verify algorithm's corectness
+            final SharedTreeNode rootNode = sharedTreeSubgraph.rootNode;
```

Comment:
Does the tree need to be hardcoded in the test?

You are building a dynamic model - GBM can produce a different (better :)) result in the future. Why would this test fail as a result of a good change?

I would traverse both trees and make sure they are the same.

Reply:
Nope. Changed.